File: Python/Easy/Filter/String/1957. Delete Characters to Make Fancy String.py
def makeFancyString(s: str) -> str:
    result = ''
    count = 0

    for i in range(0, len(s)):
        if s[i] == s[i - 1]:
            count += 1
        else:
            count = 1

        if count < 3:
            result += s[i]

    return result


s = "leeetcode"
# s = "aaabaaaa"
# s = "aab"
print(makeFancyString(s))

File: PHP/Easy/Filters/String/1957_Delete_Characters_to_Make_Fancy_String.php
<?php
function makeFancyString($s)
{

    $result = '';
    $count = 0;
    for ($i = 0; $i < strlen($s); $i++) {
        if ($s[$i] == $s[$i - 1]) {
            $count++;
        } else {
            $count = 1;
        }
        if ($count < 3) {
            $result .= $s[$i];
        }
    }
    return $result;
}